core/commands/log.go

```diff
@@ -101,34 +104,14 @@ Outputs event log messages (not other log messages) as they are generated.
 `,
 	},
 
-	Run: func(req cmds.Request, res cmds.Response) {
-		ctx := req.Context()
+	Run: func(req *cmds.Request, res cmds.ResponseEmitter, env cmds.Environment) error {
+		ctx := req.Context
 		r, w := io.Pipe()
 		go func() {
 			defer w.Close()
 			<-ctx.Done()
 		}()
 		lwriter.WriterGroup.AddWriter(w)
-		res.SetOutput(r)
+		return res.Emit(r)
```

Comment:
Same.

Reply:
@Stebalien Based on https://github.com/ipfs/go-ipfs/pull/5665#discussion_r229153266.

`r` is a reader here. Should we use `res.Emit(r)` directly?

Reply:
You're right.